tests/functional/object/mcg/test_noobaa_bucket_error_state_mode.py

```
@@ -0,0 +1,560 @@
+            if check_not_optimal and mode != "OPTIMAL":
+                log.info(f"Bucket {bucket_name} entered non-OPTIMAL mode: {mode}")
+                return mode
+            elif not check_not_optimal and mode == expected_mode:
+                log.info(f"Bucket {bucket_name} entered expected mode: {mode}")
+                return mode
+    except TimeoutExpiredError:
+        if check_not_optimal:
+            assert False, (
+                f"Bucket {bucket_name} is still OPTIMAL after "
+                f"{BUCKET_MODE_TIMEOUT}s (last mode: {last_mode})"
+            )
```

Ruff (0.15.15)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)